AEPMessaging/Sources/Messaging+PublicAPI.swift

```diff
@@ -54,33 +50,41 @@ import UserNotifications
                           source: EventSource.requestContent,
                           data: eventData)
         MobileCore.dispatch(event: event)
+        return true
     }
 
     /// Sends the push notification interactions as an experience event to Adobe Experience Edge.
     /// - Parameter response: UNNotificationResponse object which contains the payload and xdm informations.
-    static func handleNotificationResponse(_ response: UNNotificationResponse) {
-        hasApplicationOpenedForResponse(response, completion: { isAppOpened in
+    /// - Returns: boolean value that signifies whether the notification originated from AJO and whether the API has proceeded with notification tracking.
+    @discardableResult
+    static func handleNotificationResponse(_ response: UNNotificationResponse) -> Bool {
+        let notificationRequest = response.notification.request
+
+        // Checking if the message has the _xdm key that contains tracking information
+        let xdm = notificationRequest.content.userInfo[MessagingConstants.XDM.AdobeKeys._XDM] as? [String: Any]
+        if xdm == nil {
+            Log.debug(label: MessagingConstants.LOG_TAG, "XDM specific fields are missing from push notification response. Ignoring to track push notification.")
+            return false
+        }
```

Comment:
generally speaking, we prefer to use a `guard` statement when the block will result in returning from the method.  same change can be made in deprecated method above too.

Reply:
right, we need to add that to the check as well